### ISSUE TITLE
bug: db.get() catch or then not firing

### ISSUE THREAD

**Issue by dwilt** [pouchdb/pouchdb#4950]
We're currently building out an Ionic application and we have a simple `get` that we're running in the beginning `.run` of our app. We seem to have some kind of race condition because this doesn't always happen. About 80% of the time, the `get` works. We literally can hit stop in Xcode to stop the app, will hit play and re-run the exact build we have, and 20% of the time it will completely hang - the `.then` or `.catch` will just not fire and the rest of the app is held up. I'll post the code, but it's pretty basic:

```javascript
this.db = new PouchDB('db-config');

this.dbconfig.get(this.configdoc)
    .then((doc) => {
        // this never fires
    })
    .catch(error => {
        // and this never fires
    });
```

Any ideas?

**Comment by nolanlawson** [comment]
Is this inside of iOS using WKWebView? PouchDB is currently broken in that environment; you have to use the FruitDOWN adapter as a workaround.

**Comment by dwilt** [comment]
No, it's not.

**Comment by dwilt** [comment]
Atleast, I'm pretty sure it's not. Is it possible that too many calls to pouch are being made at the same time (even if it's to different db's) that pouch would just collapse?

**Comment by mnewmedia** [comment]
Is the sqlite plugin loaded?
Did you wrap it inside:
```js
$ionicPlatform.ready(function () {
});
```

**Comment by dwilt** [comment]
@mnewmedia yes, we're using `cordova-sqllite-storage`. And yes, we actually have our entire application inside of `$ionicPlatform.ready` as we have a bootstrap file that waits for that event and then loads our main `app.js` to ensure that everything is in that event.

**Comment by mnewmedia** [comment]
Ok, but I also had exactly the same issue once, then finally i found out, the plugin did not load always.

Maybe you can try the following code snippet. And if the `get` doesn't fire you should see the plugin error printed out..
```js
window.sqlitePlugin.echoTest(function () {
    console.log('plugin loaded');
}, function () {
    console.log('plugin error');
});
```

**Comment by dwilt** [comment]
I will try the `echoTest` on the plugin and get back to you. Thank you for the suggestion.

We've logged out the db info initially using:
```javascript
PouchDB.debug.enable('*');
```

When we log the `get` in the console for example, we do get a promise object back. It's definitely not a fault of the get because it works _sometimes_.
